engine/job.go

```diff
@@ -248,3 +255,15 @@ func (job *Job) StatusCode() int {
 func (job *Job) SetCloseIO(val bool) {
 	job.closeIO = val
 }
+
+// When called, causes the Job.WaitCancelled channel to unblock.
+func (job *Job) Cancel() {
+	job.cancelOnce.Do(func() {
```

Comment:
Yes. However, I think this is the wrong place to put it, because while yes, the only use of `Cancel` is if a client disconnects right now is when the http connection is lost, in the future there may be other reasons to cancel. So I've put the message at the point of connection loss (i.e, the caller of `Cancel`.)